src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,56 @@
+import { FC } from 'react';
+import { getPreparedTodos } from '../../utils/getPreparedTodos';
+import { Todo } from '../../types/Todo';
+import { FilterType } from '../../types/FilterType';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+interface TodoListProps {
+  todos: Todo[];
+  filter: FilterType;
+  onDeleteTodo: (todoId: Todo['id']) => Promise<void>;
+  tempTodo: Todo | null;
+  processingTodoIds: Todo['id'][];
+  onToggleStatusSingleTodo: (
+    todoId: Todo['id'],
+    completed: boolean,
+  ) => Promise<void>;
+  onUpdateTodo: (todoId: number, title: string) => Promise<void>;
+}
+
+export const TodoList: FC<TodoListProps> = ({
+  todos,
+  filter,
+  onDeleteTodo,
+  tempTodo,
+  processingTodoIds,
+  onToggleStatusSingleTodo,
+  onUpdateTodo,
+}) => {
+  const visibleTodos = getPreparedTodos<Todo>(todos, filter);
```

Comment:
```suggestion
  const visibleTodos = getPreparedTodos(todos, filter);
```
better add type to function return